service/controller/resource/cphostedzone/resource.go

```diff
@@ -61,68 +63,90 @@ func (r *Resource) addHostedZoneInfoToContext(ctx context.Context, cr infrastruc
 	}
 
 	{
-		hostedZoneID, err := r.lookup(ctx, cc.Client.ControlPlane.AWS.Route53, cr)
+		cpHostedZoneID, cpInternalHostedZoneID, err := r.lookup(ctx, cc.Client.ControlPlane.AWS.Route53, cr)
 		if err != nil {
 			return microerror.Mask(err)
 		}
 
-		cc.Status.ControlPlane.HostedZone.ID = hostedZoneID
+		cc.Status.ControlPlane.HostedZone.ID = cpHostedZoneID
+		cc.Status.ControlPlane.InternalHostedZone.ID = cpInternalHostedZoneID
 	}
 
 	return nil
 }
 
-func (r *Resource) lookup(ctx context.Context, client Route53, cr infrastructurev1alpha2.AWSCluster) (string, error) {
+func (r *Resource) lookup(ctx context.Context, client Route53, cr infrastructurev1alpha2.AWSCluster) (string, string, error) {
```

Comment:
I'd add named return values because it becomes cryptic

```suggestion
func (r *Resource) lookup(ctx context.Context, client Route53, cr infrastructurev1alpha2.AWSCluster) (cpHostedZoneID, cpInternalHostedZoneID string, error) {
```